elements/pf-timestamp/pf-timestamp.ts

```diff
@@ -45,72 +47,34 @@ export class PfTimestamp extends LitElement {
 
   @property({ reflect: true, attribute: 'hour-12', converter: BooleanStringConverter }) hour12?: boolean;
 
-  #date = new Date();
-
-  #isoString = this.#date.toISOString();
-
   get isoString() {
-    return this.#isoString;
+    return this.#timestamp.isoString;
   }
 
   get time() {
-    const { hour12, customFormat, dateFormat: dateStyle, timeFormat: timeStyle, utc } = this;
-    const timeZone = utc ? 'UTC' : undefined;
-    const formatOptions = customFormat || { hour12, dateStyle, timeStyle, timeZone };
-    const formattedDate = this.#date.toLocaleString(this.locale, formatOptions);
-    return this.relative ? this.#getTimeRelative(this.#date) : `${formattedDate}${this.displaySuffix ? ` ${this.displaySuffix}` : ''}`;
+    return this.#timestamp.time;
   }
 
-  willUpdate() {
+  #timestamp = new TimestampController(this);
+
+  willUpdate(changedProperties: Map<string, any>) {
     if (!this.displaySuffix && this.utc) {
       this.displaySuffix = 'UTC';
     }
+
+    for (const [prop] of changedProperties) {
+      if (this.#timestamp.isOption(prop)) {
+        // @ts-ignore
```

Comment:
@bennypowers could not get over this TS error for the life of me. Incorporated your feedback, although I made `options` public and defined the accessor methods on that property instead.

Reply:
All good, Ivana.

I fixed up the typescript errors by taking a slightly different tac with the controller APIs. Some of the tests are still failing, could just be config defaults :shrug: take a look when you get a minute. 

Thanks for working on this